lib/instance.js

```diff
@@ -742,7 +743,16 @@ Instance.prototype.reload = function(options) {
     include: this.$options.include || null
   });
 
-  return this.Model.findOne(options).bind(this).then(function(reload) {
+  return this.Model.findOne(options).bind(this)
+  .tap(function (reload) {
+    if (!reload) {
+      throw new sequelizeErrors.InstanceError('Instance cannot be reloaded because does not exist in the database');
```

Comment:
`Instance could not be reloaded because it does not exist anymore (find call returned null)`